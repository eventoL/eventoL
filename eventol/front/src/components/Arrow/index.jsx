import React from 'react';
import PropTypes from 'prop-types';

import Logger from '../../utils/logger';

export default class Arrow extends React.PureComponent {
  static propTypes = {
    onClick: PropTypes.func,
    style: PropTypes.shape({}),
    type: PropTypes.oneOf(['prev', 'next']).isRequired,
  };

  static defaultProps = {
    onClick: Logger.log,
    style: {},
  };

  getClassNameByType() {
    const {type} = this.props;
    if (type === 'prev') {
      return 'arrow-prev fa fa-chevron-left';
    }
    if (type === 'next') {
      return 'arrow-next fa fa-chevron-right';
    }
    return '';
  }

  handleOnClick = event => {
    event.preventDefault();
    event.stopPropagation();
    const {onClick} = this.props;
    onClick();
  };

  render() {
    const {style} = this.props;
    return (
      <div
        className={`arrow ${this.getClassNameByType()}`}
        onClick={this.handleOnClick}
        onKeyPress={this.handleOnClick}
        role="button"
        style={{...style, display: 'block', color: 'black'}}
        tabIndex="0"
      />
    );
  }
}
