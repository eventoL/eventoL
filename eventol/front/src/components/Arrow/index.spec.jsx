import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('../../utils/logger', () => ({log: jest.fn()}));

import Arrow from '.';

describe('Arrow', () => {
  let component;
  let tree;
  let onClickSpy;
  let event;

  beforeEach(() => {
    onClickSpy = jest.fn();
    event = {
      preventDefault: jest.fn(),
      stopPropagation: jest.fn(),
    };
    component = renderer.create(<Arrow onClick={onClickSpy} type="prev" />);
    tree = component.toJSON();
  });

  describe('Snapshots', () => {
    test('Default', () => {
      expect(tree).toMatchSnapshot();
    });

    test('prev', () => {
      component = renderer.create(<Arrow type="prev" />);
      tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('next', () => {
      component = renderer.create(<Arrow type="next" />);
      tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('another type', () => {
      component = renderer.create(<Arrow type="another" />);
      tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('with custom style', () => {
      component = renderer.create(
        <Arrow style={{padding: '10px'}} type="prev" />
      );
      tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  test('should call onClick handle', () => {
    tree.props.onClick(event);
    expect(event.preventDefault).toBeCalled();
    expect(event.stopPropagation).toBeCalled();
    expect(onClickSpy).toBeCalled();
  });
});
