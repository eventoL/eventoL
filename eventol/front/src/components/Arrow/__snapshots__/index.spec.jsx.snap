// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow Snapshots Default 1`] = `
<div
  className="arrow arrow-prev fa fa-chevron-left"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  style={
    Object {
      "color": "black",
      "display": "block",
    }
  }
  tabIndex="0"
/>
`;

exports[`Arrow Snapshots another type 1`] = `
<div
  className="arrow "
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  style={
    Object {
      "color": "black",
      "display": "block",
    }
  }
  tabIndex="0"
/>
`;

exports[`Arrow Snapshots next 1`] = `
<div
  className="arrow arrow-next fa fa-chevron-right"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  style={
    Object {
      "color": "black",
      "display": "block",
    }
  }
  tabIndex="0"
/>
`;

exports[`Arrow Snapshots prev 1`] = `
<div
  className="arrow arrow-prev fa fa-chevron-left"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  style={
    Object {
      "color": "black",
      "display": "block",
    }
  }
  tabIndex="0"
/>
`;

exports[`Arrow Snapshots with custom style 1`] = `
<div
  className="arrow arrow-prev fa fa-chevron-left"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  style={
    Object {
      "color": "black",
      "display": "block",
      "padding": "10px",
    }
  }
  tabIndex="0"
/>
`;
